iocore/eventsystem/unit-tests/test_MIOBufferWriter.cc

```diff
@@ -0,0 +1,217 @@
+/** @file
```

Comment:
Can you move this around it parallels the `lib/ts` structure? That is, an explicit `main` file which is then linked with the unit test files in to the `test_eventsystem` binary.

Reply:
I don't see the advantage in globing independent unit tests together into one big horking executable.  Automake's 'check' target will run all of the unit tests in a directory, if that's what you want to do.   But anyway.

Reply:
Two reasons:
* It makes running selected tests overall easier because there is only one executable and it can be told (via tags) to run precise subsets of the tests.
* Overall compile time is reduced. Building the base Catch code takes a noticeable amount of compile time. This is amortized over all the tests if they are grouped.

Reply:
* But we don't need subsets if they are not glommed together in the first place.
* That's only true when doing regression testing.  And, only if you are running all unit tests as a regression test.  Which, ideally, we would not do, we'd run only unit tests which depend on changed source files.

Reply:
The current CI always runs all regression and units tests.

Reply:
Yeah it's so wonderful.

Reply:
I agree with @ywkaras I think it is a bad idea to put all the unit tests into one binary.  

- Having multiple binaries allows it is run in parallel. 
- The compile time can be done in parallel if they are separate tests and will be a small amount of time compared to running the tests.
- You can still selectively run tests choosing the binary and tags.